TESTING.md

```diff
@@ -0,0 +1,26 @@
+# TESTING mypy
+The basic way to run all tests:
+
+    $ pip3 install -r test-requirements.txt
+    $ python2 -m pip install -U typing
+    $ ./runtests.py
+
+Mypy is tested using `pytest`. This can be installed as shown above, using `test-requirements.txt`. In essence, there are multiple kinds of tests for mypy:
+- Unit tests (Found under `./mypy/test/`, [example](mypy/test/testapi.py))
+- Data-driven unit and integration tests (Found under `./mypy/test/`, [example pep561](mypy/test/testpep561.py))
+
+As might be clear, some tests are driven by data. For now, this data resides in `./test-data/unit/`. Do note: this syntax of these data-driven tests is parsed by Python test-cases in `./mypy/test/` by specifying the `files=[]` property. Some test files specify their own conditions on the tests (input data, flags, etc.). For more information about the syntax, read the next section.
```

Comment:
I'd drop the "For now," -- I don't think anyone is thinking about moving it elsewhere.

Also:

- "this syntax" -> "the syntax"
- "specifying the `files=[]` property" -> "setting the `files` class attribute to a list of filenames"
- I don't actually see anything in the next section about the syntax of the test data files; it basically just contains a link. Maybe you can drop that section and just plant the link here?